### PR TITLE
feat: cost dashboard date filters & CSV download

### PR DESCRIPTION
**Pull request by juleskuehn**

### CODE REVIEW

File: django/chat/llm.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def _get_llm(self) -> AzureOpenAI:
 
     def _get_embed_model(self) -> AzureOpenAIEmbedding | MockEmbedding:
         if self.mock_embedding:
-            return MockEmbedding(1536, callback_manager=self._callback_manager)
+            return MockEmbedding(1536)
```
Comment (smcdonald-jus) [inline django/chat/llm.py:268]: Guessing the callback_manager argument was found to be redundant or something?

File: django/otto/views.py [hunk 2]
```diff
@@ -670,9 +679,83 @@ def cost_dashboard(request):
...
+    start_date = request.GET.get("start_date", None)
+    end_date = request.GET.get("end_date", None)
 
-    # First, filter the costs by the selected options
     raw_costs = Cost.objects.all()
+
+    # Filter by dates
+    if date_group == "last_90_days":
+        start_date = timezone.now().date() - timedelta(days=90)
+        end_date = timezone.now().date() - timedelta(days=1)
```
Comment (smcdonald-jus) [inline django/otto/views.py:691]: What's the rationale behind excluding the current date from the "last N days" date groups?

Comment (juleskuehn) [reply django/otto/views.py:691]: The current day is incomplete. So it gives you a slightly false idea of what "a week" costs.

Comment (smcdonald-jus) [reply django/otto/views.py:691]: Fair enough!

**Review by smcdonald-jus — APPROVED** [review]

**Pull request merged**